test/langtools/tools/javac/processing/options/TestNoteOnImplicitProcessing.java

```diff
@@ -0,0 +1,274 @@
+ * $ javac -cp ImplicitProcTestProc.jar -Xlint:-options                     HelloWorldTest.java
+ * $ javac -cp ImplicitProcTestProc.jar -Xlint:none                         HelloWorldTest.java
+ */
+
+public class TestNoteOnImplicitProcessing extends TestRunner {
+    public static void main(String... args) throws Exception {
+
+        var self  = new TestNoteOnImplicitProcessing();
+        Path jarFilePath = self.createProcessorJarFile();
+        self.runTests(m -> new Object[] { Paths.get(m.getName()), jarFilePath });
+    }
+
+    private ToolBox tb = new ToolBox();
+    private String processorName = "ImplicitProcTestProc";
+
+    public TestNoteOnImplicitProcessing() {
+        super(System.err);
+    }
+
+    private Path createProcessorJarFile() throws Exception {
+        // Write out shared-use source file
+        tb.writeFile("HelloWorldTest.java",
+                     """
+                     public class HelloWorldTest {
+                         public static void main(String... args) {
+                             System.out.println("Hello world test.");
+                         }
+                     }
+                     """);
+
+        JarTask jarTask = new JarTask(tb, processorName + ".jar");
+        Path apDir = Paths.get(".");
+
+        // write out META-INF/services file for the processor
+        Path servicesFile =
+            apDir
+            .resolve("META-INF")
+            .resolve("services")
+            .resolve(Processor.class.getCanonicalName());
+        tb.writeFile(servicesFile,
+                     processorName);
+
+        // write out processor source file
+        tb.writeFile(processorName + ".java",
+                     """
+                     import java.util.Set;
+                     import javax.annotation.processing.*;
+                     import javax.lang.model.SourceVersion;
+                     import javax.lang.model.element.TypeElement;
+
+                     @SupportedAnnotationTypes("*")
+                     public class ImplicitProcTestProc extends AbstractProcessor {
+                         public ImplicitProcTestProc() {super();}
+
+                         @Override
+                         public boolean process(Set<? extends TypeElement> annotations,
+                                                RoundEnvironment roundEnv) {
+                             if (roundEnv.processingOver()) {
+                                 System.out.println("ImplicitProcTestProc run");
+                             }
+                             return true;
+                         }
+
+                         @Override
+                         public SourceVersion getSupportedSourceVersion() {
+                             return SourceVersion.latest();
+                         }
+                     }
+                     """);
+
+        // Compile the processor
+        new JavacTask(tb)
+            .files(processorName + ".java")
```

Comment:
there's a utility method `ToolBox.findJavaFiles` which is generally useful in this sort of context